lib/iris/common/metadata.py

```diff
@@ -140,6 +140,88 @@ def __new__(mcs, name, bases, namespace):
         return super().__new__(mcs, name, bases, namespace)
 
 
+#
+# Dictionary operations for dealing with the CubeAttrsDict "split"-style attribute
+# dictionaries.
+#
+# The idea here is to convert a split-dictionary into a "plain" one for calculations,
+# whose keys are all pairs of the form ('global', <keyname>) or ('local', <keyname>).
+# And to convert back again after the operation, if the result is a dictionary.
+#
+# For "strict" operations this clearly does all that is needed.  For lenient ones,
+# we _might_ want for local+global attributes of the same name to interact.
+# However, on careful consideration, it seems that this is not actually desirable for
+# any of the common-metadata operations.
+# So, we simply treat "global" and "local" attributes of the same name as entirely
+# independent. Which happily is also the easiest to code, and to explain.
+#
+def xd_is_split(dic):
```

Comment:
Considerations for 'packaging' of these convenience functions:

- `iris.common.metadata` is a public module. Do we definitely want these functions to be public? They have a very specific purpose.
- Would they be better in their own module since they are all related and distinct from everything that follows?
  e.g. `_cube_attr_handling.py`
- Can you explain the `xd_` naming convention? I don't get it!

Reply:
OK I have now re-worked all this to its own module, simplified + (hopefully) clarified.
https://github.com/SciTools/iris/pull/5538/commits/8d61d9b42c208c7a81b8332f961bb9b3fff1a1b7